client/host_queue_test.go

```diff
@@ -678,6 +1175,46 @@ func testHostQueueFetchBatches(
 	closeWg.Wait()
 }
 
+func testWriteTaggedOp(
+	namespace string,
+	id string,
+	tags map[string]string,
+	value float64,
+	timestamp int64,
+	timeType rpc.TimeType,
+	completionFn completionFn,
+) *writeTaggedOperation {
+	w := &writeTaggedOperation{}
+	w.reset()
+	w.namespace = ident.StringID(namespace)
+	w.request.ID = []byte(id)
+	w.request.Datapoint = &rpc.Datapoint{
+		Value:             value,
+		Timestamp:         timestamp,
+		TimestampTimeType: timeType,
+	}
+	w.request.EncodedTags = testEncode(tags)
+	w.completionFn = completionFn
+	return w
+}
+
+func testEncode(tags map[string]string) []byte {
```

Comment:
Hm interesting, don't want to use a real tag encoder here like the one in the PR #479?

Reply:
Nah explicitly chose to use mocks/stubbed code to keep the package level unit tests separate.